Fix Certbot's Apache plugin doesn't work on Scientific Linux

tyborr opened the pull request:
This PR adds OVERRIDE_CLASS in certbot-apache/entrypoint.py for Scientific Linux. Fixes #7248

ohemorange commented:
@bmw, I see you created #7248 and may therefore have more context on this problem.

bmw commented:
@tyborr, how did you come up with "scientific" as the string here? Do you have a Scientific Linux system to test on?

If so, I'm curious what the output of:
```
from certbot.util import get_os_info, get_systemd_os_like
print(get_os_info())
print(get_systemd_os_like())
```
is on that system.

https://community.letsencrypt.org/t/noinstallationerror-cannot-find-apache-executable-apache2ctl/97980/4 suggests it should be "scientific linux".\

tyborr commented:
@bmw I setup Docker container to verify what proper values should be. As far as I understood those two methods that you mentioned retrieve data from `/etc/os-release`, `ID` and `ID_LIKE` vars, which look like this:
```
[root@026bbd17c8b2 ~]# cat /etc/os-release
NAME="Scientific Linux"
VERSION="7.6 (Nitrogen)"
ID="scientific"
ID_LIKE="rhel centos fedora"
VERSION_ID="7.6"
PRETTY_NAME="Scientific Linux 7.6 (Nitrogen)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:scientificlinux:scientificlinux:7.6:GA"
HOME_URL="http://www.scientificlinux.org//"
BUG_REPORT_URL="mailto:scientific-linux-devel@listserv.fnal.gov"

REDHAT_BUGZILLA_PRODUCT="Scientific Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.6
REDHAT_SUPPORT_PRODUCT="Scientific Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.6"
```

tyborr commented:
My bad - didn't read the code in `entrypoint.sh` carefully enough. This line `override_class = OVERRIDE_CLASSES[os_name]` checks for os_name, not id_like. Fixed in my last commit.

tyborr commented:
I added another key in `entrypoint.sh` and updated `CHANGELOG.md` per your request. Unfortunately I'm afraid I messed up this PR when I was trying to rebase working branch. Let me know if you need me to re-do this whole PR from scratch.

bmw commented on CHANGELOG.md:
```diff
@@ -16,7 +16,7 @@ Certbot adheres to [Semantic Versioning](https://semver.org/).
 ### Fixed
 
-*
+* Fixed OS detection in the Apache plugin on Scientific Linux."
```
```suggestion
* Fixed OS detection in the Apache plugin on Scientific Linux.
```

tyborr commented:
Fixed
